src/PrestaShopBundle/Form/Admin/Sell/Product/Shipping/ShippingType.php

```diff
@@ -110,18 +116,35 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                 'default_empty_data' => 0.0,
                 'modify_all_shops' => true,
             ])
-            ->add('carriers', ChoiceType::class, [
-                'choices' => $this->carrierChoiceProvider->getChoices(),
+            // related all shops checkbox is rendered in javascript side, but it is still needed here to be correctly filled with data
+            ->add($modifyAllShopsInputName, HiddenType::class)
+        ;
+
+        $builder->addEventListener(FormEvents::PRE_SET_DATA, function (FormEvent $event) use ($modifyAllShopsInputName) {
+            $carrierChoices = $this->carrierChoiceProvider->getChoices();
+            $form = $event->getForm();
+            // Vue js CarrierSelector component mounts on this one instead of rendering default bootstrap choices list
+            $form->add('carriers', ChoiceType::class, [
+                'choices' => $carrierChoices,
+                'attr' => [
+                    'data-modify-all-shops-name' => sprintf('product[shipping][%s]', $modifyAllShopsInputName),
+                    'data-choice-input-name' => 'product[shipping][carriers][]',
```

Comment:
These data attributes are not needed as well, you should get the name of the select field instead this way it's really dynamic and will adapt if it's renamed or moved

Reply:
The modify for all is more a special case indeed, but I don't understand why the regular way doesn't work here? Seems like it could be completely out of the Vue component no?

Reply:
using the regular input (by form option modify_all_shops ž true) when you click on the button which supposed to expand carriers selection, instead of expanding it and it kind of works strange then. Btw it for some reason renders above the component (though I guess that wouldn't be hard to fix)

https://user-images.githubusercontent.com/31609858/216284316-544a1e42-a59d-4ba5-b9d5-0e67f35adc9e.mov

Reply:
altough with custom one, the multishop checkbox always stays visible :/